src/core_plugins/kibana/common/field_formats/types/url.js

```diff
@@ -106,7 +106,9 @@ UrlFormat.prototype._convert = {
           linkLabel = label;
         }
 
-        return `<a href="${url}" target="_blank">${linkLabel}</a>`;
+        const linkTarget = this.param('openLinkInCurrentTab') ? '_self' : '_blank';
+
+        return `<a href="${url}" target="${linkTarget}">${linkLabel}</a>`;
```

Comment:
I think we should add [`rel=noopener`](https://mathiasbynens.github.io/rel-noopener/) since this may contain external links.